lib/commands/location.js

```diff
@@ -0,0 +1,29 @@
+import { exec } from 'teen_process';
+import { fs, util } from 'appium-support';
+import log from '../logger';
+
+let commands = {};
+
+commands.setGeoLocation = async function (location) {
+  let {latitude, longitude} = location;
+
+  if (!util.hasValue(latitude) || !util.hasValue(longitude)) {
+    log.errorAndThrow(`Both latitude and longitude should be set`);
+  }
+
+  if (this.isRealDevice()) {
+    if (!(await fs.which('idevicelocation'))) {
+      log.errorAndThrow(`idevicelocation doesn't exist on the host`);
+    }
+    try {
+      await exec('idevicelocation', ['-u', this.opts.udid, `${latitude}`, `${longitude}`]);
```

Comment:
```
return await exec('idevicelocation', ['-u', this.opts.udid, `${latitude}`, `${longitude}`]);
```

so then `else` is not needed

Reply:
In the past, this method only returned null. If I do `return await exec('idevicelocation', ['-u', this.opts.udid, `${latitude}`, `${longitude}`]);` then we will return the output of exec. Is that something we want?

Reply:
is the returned result checked anywhere? If not then the change is ok. If not then you can simply add `return` after `exec`

Reply:
Do you think that looks better than if else? The if else brings distinction in this case for real devices and simulators. On top of that it prevents us putting a ugly looking return;

Reply:
it was just a suggestion. 
Also, please don't forget to sign the CLA (make sure your email in the local git profile is correct).

Reply:
I prefer keeping the if/else structure to highlight the control flow. If we wanted to get rid of it completely, the  `setGeolocation` function could be added to the ios-deploy object, so this whole function would just call 'this.opts.device.setGeolocation(`${latitude}`, `${longitude}`);'. That is unnecessary at the moment, I think.